api/gateway/basecommerce/basecommerce.cfc

```diff
@@ -0,0 +1,281 @@
+						local.response.setTransactionID(local.results.transactionid);
+					}
+					if(structKeyExists(local.results, 'authcode')) {
+						local.response.setAuthorization(local.results.authcode);
+					}
+					if(structKeyExists(local.results, 'customer_vault_id')) {
+						local.response.setTokenID(local.results.customer_vault_id);
+					}
+					
+					//handle common 'success' fields
+					if(structKeyExists(local.results, 'avsresponse')) {
+						local.response.setAVSCode(local.results.avsresponse);
+					}
+					if(structKeyExists(local.results, 'cvvresponse')) {
+						local.response.setCVVCode(local.results.cvvresponse);
+					}
+	
+					//see if the response was successful
+					if(local.results.response EQ '1') {
+						local.response.setStatus(getService().getStatusSuccessful());
+					} else if(local.results.response EQ '2') {
+						local.response.setStatus(getService().getStatusDeclined());
+					} else {
+						//only other known state is 3 meaning, 'error in transaction data or system error'
+						local.response.setStatus(getService().getStatusFailure());
+					}
+				}
+			} else {
+				//This is bad, because Braintree didn't return a response.  Uh oh!
+				local.response.setStatus(getService().getStatusUnknown());
+			}
+		}
+		
+		return local.response;
+	}
+
+	private any function createResponse() {
+		return createObject("component", "cfpayment.api.gateway.basecommerce.response").init(argumentCollection = arguments, service = getService());
+	}
+	
+	private any function CreateEmptyResponse(struct payload=structNew()) {
+		local.response = structNew();
+		
+		local.response.message = '';
+		local.response.result = '';
+		local.response.status = 0;
+		local.response.statusCode = '';
+		local.response.testMode = true;
+		local.response.requestData = structNew();
+		local.response.requestData['gateway_url'] = '';
+		local.response.requestData['headers'] = structNew();
+		local.response.requestData['http_method'] = '';
+		local.response.requestData['payload'] = arguments.payload;
+		
+		return local.response;
+	}
+
+	//Implement primary methods
+	public any function purchase(required any money, any account, struct options=structNew()) {
+		local.argumentCollection = structNew();
+		local.argumentCollection.money = arguments.money;
+		if(isDefined('arguments.account')) local.argumentCollection.money = arguments.account;
+		local.argumentCollection.options = arguments.options;
+		local.argumentCollection.options.batType = 'XS_BAT_TYPE_CREDIT';
+		return transaction(argumentCollection = local.argumentCollection);
+	}
+	
+	public any function credit(required any money, any account, struct options=structNew()) {
+		local.argumentCollection = structNew();
+		local.argumentCollection.money = arguments.money;
+		if(isDefined('arguments.account')) local.argumentCollection.money = arguments.account;
+		local.argumentCollection.options = arguments.options;
+		local.argumentCollection.options.batType = 'XS_BAT_TYPE_DEBIT';
```

Comment:
these are backwards... where debit/credit should be reversed.  Before swapping this, please add a unit test assertion that catches this.  The value returned should be different in some way if we charge someone vs. send them money and that's something the test should look for.